src/lib/operators/table_scan/column_vs_column_table_scan_impl.cpp

```diff
@@ -28,66 +29,102 @@ std::string ColumnVsColumnTableScanImpl::description() const { return "ColumnVsC
-    segment_with_iterators(*right_segment, [&](auto right_it, const auto right_end) {
-      using LeftSegmentIterableType = typename decltype(left_it)::IterableType;
-      using RightSegmentIterableType = typename decltype(right_it)::IterableType;
-
-      using LeftType = typename decltype(left_it)::ValueType;
-      using RightType = typename decltype(right_it)::ValueType;
-
-      /**
-       * The following generic lambda is instantiated for each combinations of type (int, long, etc.) and
-       * each segment iterator type (value, value-non-null, dictionary-simd, ...)!
-       * However, not all combinations are valid or possible.
-       * Only data segments (value, dictionary) or reference segments will be compared, as a table with both data and
-       * reference segments is ruled out. Moreover it is not possible to compare strings to any of the four numerical
-       * data types. Therefore, we need to check for these cases and exclude them via the constexpr-if which
-       * reduces the number of combinations.
-       */
-
-      constexpr auto LEFT_IS_REFERENCE_SEGMENT =
-          std::is_same<LeftSegmentIterableType, ReferenceSegmentIterable<LeftType>>{};
-      constexpr auto RIGHT_IS_REFERENCE_SEGMENT =
-          std::is_same<RightSegmentIterableType, ReferenceSegmentIterable<RightType>>{};
-
-      constexpr auto NEITHER_IS_REFERENCE_SEGMENT = !LEFT_IS_REFERENCE_SEGMENT && !RIGHT_IS_REFERENCE_SEGMENT;
-      constexpr auto BOTH_ARE_REFERENCE_SEGMENTS = LEFT_IS_REFERENCE_SEGMENT && RIGHT_IS_REFERENCE_SEGMENT;
-
-      constexpr auto LEFT_IS_STRING_COLUMN = (std::is_same<LeftType, std::string>{});
-      constexpr auto RIGHT_IS_STRING_COLUMN = (std::is_same<RightType, std::string>{});
-
-      constexpr auto NEITHER_IS_STRING_COLUMN = !LEFT_IS_STRING_COLUMN && !RIGHT_IS_STRING_COLUMN;
-      constexpr auto BOTH_ARE_STRING_COLUMNS = LEFT_IS_STRING_COLUMN && RIGHT_IS_STRING_COLUMN;
-
-      if constexpr ((NEITHER_IS_REFERENCE_SEGMENT || BOTH_ARE_REFERENCE_SEGMENTS) &&
-                    (NEITHER_IS_STRING_COLUMN || BOTH_ARE_STRING_COLUMNS)) {
-        // Dirty hack to avoid https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86740
-        const auto left_it_copy = left_it;
-        const auto left_end_copy = left_end;
-        const auto right_it_copy = right_it;
-        const auto chunk_id_copy = chunk_id;
-        const auto& matches_out_ref = matches_out;
-        const auto condition = _predicate_condition;
-
-        with_comparator(condition, [&](auto predicate_comparator) {
-          auto comparator = [predicate_comparator](const auto& left, const auto& right) {
+  using LeftType = typename LeftIterable::ValueType;
+  using RightType = typename RightIterable::ValueType;
+
+  // C++ cannot compare strings and non-strings out of the box:
+  if constexpr (std::is_same_v<LeftType, std::string> == std::is_same_v<RightType, std::string>) {
+    bool flipped = false;
```

Comment:
I know it's minor, but if you'd do the flipping outside of thing function, there wouldn't be a need for vaguely named variables "flipped" (what exactly is flipped?) and "condition" (how this is different from "_predicate_condition" is not immediately obvious)

Reply:
Not sure how that improves things. We wouldn't have those variables, yes, but I would expect readers to be more surprised if an outside function magically flipped their predicate. Seeing the flip and its effect in close proximity seems to be better to me. I agree about the naming though, maybe `condition_was_flipped` and `potentially_flipped_condition`?

Reply:
yeah, at the moment I feel like awkward speaking names are better than short imprecise ones.